### PR TITLE
refactor: overhaul admin ux and ui

### PR DESCRIPTION
**Pull request by tylerslaton**
This includes a lot of changes to the admin UI as
per a big batch of feedback from @ibuildthecloud.

The main changes are:
- ToolGrid component has been updated to:
    * Sort alphabetically in each category
    * Show each tool's individual icon as well as the category icon
    * Remove hover effects for badges
    * CreateNewTool button now only has reference field
    * Fix a bug where the tool grid had an empty tool for bundless categories
    * Fixed a bug with the tool tip for category headers
- Overall background of the application is now bg-background
- OAuth app dialog is now much wider
- Fix a broken link for threads breadcrumb
- Agent edit page is not very different
    * No more accordions for tools and knowledge
    * Add tool now opens a new Dialog instead of a popover
- ToolTable now has a onRowClick and disableClickPropagation prop
- Thread and agent tables now link to their respective pages

### CODE REVIEW

File: ui/admin/app/components/agent/Agent.tsx [hunk 1]
```diff
@@ -69,55 +63,34 @@ function AgentContent({ className, onRefresh }: AgentProps) {
...
+                        such as searching the web, reading files, or interacting
+                        with other systems.
+                    </TypographyP>
+                    <ToolForm
+                        agent={agentUpdates}
+                        onChange={debouncedSetAgentInfo}
+                    />
+                </div>
+
+                <div className="p-4 flex-auto space-y-4">
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/agent/Agent.tsx:82]: there goes our last stoppage of polling in knowledge :(

File: ui/admin/app/components/agent/ToolForm.tsx [hunk 2]
```diff
@@ -100,29 +95,24 @@ export function ToolForm({
                                 ))}
                             </div>
                             <div className="flex justify-end w-full my-4">
-                                <Popover>
-                                    <PopoverTrigger asChild>
+                                <Dialog>
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/agent/ToolForm.tsx:98]: This looks way better 🙌

File: ui/admin/app/components/tools/toolGrid/ToolCard.tsx [hunk 3]
```diff
@@ -66,23 +75,29 @@ export function ToolCard({ tool, onDelete }: ToolCardProps) {
...
+                        description="Are you sure you want to delete this tool reference? This action cannot be undone."
+                        onConfirm={() => onDelete(tool.id)}
+                        confirmProps={{
+                            variant: "destructive",
+                            children: "Delete",
+                        }}
+                    >
+                        <Button
+                            variant="ghost"
+                            size="sm"
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/tools/toolGrid/ToolCard.tsx]: size="icon"

File: ui/admin/app/components/tools/toolGrid/ToolGrid.tsx [hunk 4]
```diff
@@ -13,46 +14,75 @@ interface ToolGridProps {
...
+                    (tool) =>
+                        tool.name
+                            ?.toLowerCase()
+                            .includes(searchTerm.toLowerCase()) ||
+                        tool.metadata?.category
+                            ?.toLowerCase()
+                            .includes(searchTerm.toLowerCase()) ||
+                        tool.description
+                            ?.toLowerCase()
+                            .includes(searchTerm.toLowerCase())
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/tools/toolGrid/ToolGrid.tsx]: pro-tip

```ts
                        [tool.name, tool.metadata?.category, tool.description]
                            .filter(x => !!x)
                            .join("|")
                            .toLowerCase()
                            .includes(searchTerm.toLowerCase())
```
is a bit cleaner

File: ui/admin/app/components/agent/Agent.tsx (hunk 1 above)
Comment (tylerslaton) [reply ui/admin/app/components/agent/Agent.tsx:82]: RIP.

**Pull request merged**